fix: upgrade packadges

rodionchernyshov opened the pull request:
<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **Style**
  * Enforced light color-scheme and consistent white backgrounds for forms and textareas; refined color picker, tightened sidebar and builder layout for improved responsiveness and overflow handling.

* **Bug Fixes**
  * Made annotation color generation deterministic with robust fallbacks to prevent rendering glitches.

* **Chores**
  * Consolidated UI lint configuration and simplified lint scripts; upgraded and reshuffled many frontend dependencies and tooling.

* **Tests**
  * Stabilized a color-related test to handle style/value variability.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

rodionchernyshov committed:
fix: dark mode blink, grid layout, and improve color picker styling

- Fix dark mode blink by adding color-scheme meta tag and explicit colors
- Fix grid layout overflow issues with proper constraints and positioning
- Improve color picker styling to match WDS design system
- Replace @apache-arrow/ts with precompiled apache-arrow package
- Remove code comments from eslint.config.js

Fixes:
- Dark mode flash on page load
- Sidebar overflow causing content cutoff
- Textarea switching to dark mode
- Color picker missing hover/focus states

Changes:
- src/ui/index.html: Add color-scheme meta and form element styles
- src/ui/src/builder/BuilderApp.vue: Fix grid layout constraints
- src/ui/src/builder/sidebar/BuilderSidebar.vue: Add width constraints
- src/ui/src/builder/settings/BuilderFieldsColor.vue: Enhance color picker
- src/ui/src/wds/WdsTextareaInput.vue: Add explicit background color
- src/ui/package.json: Replace @apache-arrow/ts with apache-arrow
- src/ui/src/components/core/content/CoreDataframe.vue: Update imports
- src/ui/eslint.config.js: Remove code comments

rodionchernyshov committed:
Fix linting issues from CI

- Remove non-existent @typescript-eslint/ban-types disable comment
- Fix unnecessary escape character in regex pattern
- Replace single quotes with double quotes in builderEditorWorker.ts
- Replace any type with unknown in builderEditorWorker.ts

rodionchernyshov committed:
Fix CoreAnnotatedText snapshot test

- Add error handling and fallbacks to ensure getAnnotationBgColor always returns valid color
- Improve generateColorCss to use rgb format for better browser compatibility
- Add getAnnotationStyle helper to ensure style object always has backgroundColor
- Update test to check both HTML attribute and inline style property (Vue applies styles directly to element.style)
- Fix snapshot to match actual rendered output

rodionchernyshov committed:
fix: codereview improvements

rodionchernyshov committed:
Address CodeRabbit suggestions

- Use globals package for ESLint browser/node globals instead of manual declaration
- Pin volar-service-vetur to ^0.0.64 instead of wildcard version
- Add missing globals (MediaStreamConstraints, RequestInit) not in standard set

rodionchernyshov committed:
Move TypeScript from dependencies to devDependencies

TypeScript is a build/dev-time tool and should not be in runtime dependencies

rodionchernyshov committed:
Fix TypeScript 5.9 type errors

- Add missing runId and exit properties to BlueprintExecutionLog type
- Add ts-expect-error for Monaco TypeScript API compatibility issue

bybash commented on src/ui/src/builder/useDragDropComponent.ts:
```diff
@@ -23,7 +23,7 @@ If the user gets off bounds by MAX_DISTANCE_FROM_CANDIDATE_PX, the candidacy is
 const MAX_DISTANCE_FROM_CANDIDATE_PX = 30;
 const dragDropMimeRegex =
-	/^application\/json;writer=(?<componentType>\w+),(?<componentId>[\w\-]*)$/;
+	/^application\/json;writer=(?<componentType>\w+),(?<componentId>[\w-]*)$/;
```
Is this intentional or a linter fix?

rodionchernyshov replied:
This is a linter fix.

bybash commented on src/ui/src/builder/BuilderApp.vue:
```diff
@@ -561,8 +571,18 @@ onUnmounted(() => {
 	height: 100%;
 }
 
-.mainGrid--fullWidth {
```
I think we use this one, will this be covered by this `builderMainWrapper--fullWidth `?

rodionchernyshov replied:
Good catch! The code actually uses `mainGrid--fullWidth` on the `ShareResizeVertical` component, not `builderMainWrapper--fullWidth`.

bybash commented on src/ui/src/builder/BuilderApp.vue:
```diff
@@ -549,9 +550,18 @@ onUnmounted(() => {
+	max-width: 288px;
+	width: fit-content;
+	padding: 0;
+	border-radius: 0;
+	border-left: none;
+	border-top: none;
+	border-bottom: none;
```
Do we need these ones?

rodionchernyshov replied:
after upgrade I found that we need changes there and avoid UI issues so I've added some fixes and implements.

bybash commented on src/ui/src/builder/BuilderApp.vue:
```diff
@@ -561,8 +571,18 @@ onUnmounted(() => {
 
-.mainGrid--fullWidth {
-	grid-column: 1 / -1;
+.builderMainWrapper {
```
Where do we use this new class?

rodionchernyshov replied:
The `.builderMainWrapper` class is **not currently used** in the template.

bybash commented on src/ui/src/builder/BuilderApp.vue:
```diff
@@ -549,9 +550,18 @@ onUnmounted(() => {
 .sidebar {
 	grid-column: 1 / 2;
-	grid-row: 2 / 5;
+	grid-row: 2 / 3;
```
Previously, is this one set incorrectly?

rodionchernyshov replied:
Yes, `grid-row: 2 / 5` was incorrect. The grid has 3 rows, and the sidebar should only span row 2 (the main content area), not rows 2-4. The fix to `grid-row: 2 / 3` is correct and ensures the sidebar is properly constrained to the main content area.

bybash replied:
@as-flow is this something needed for shared blueprints?

as-flow replied:
I don't think so @bybash

madeindjs replied:
I guess that break preview mode & Vault UI

<img width="1717" height="1326" alt="Image" src="https://github.com/user-attachments/assets/51480362-78a9-48f1-a240-ab6758cc4789" />

<img width="1717" height="1326" alt="Image" src="https://github.com/user-attachments/assets/bcef8eaf-3233-4e5a-80e4-dad6d7c2df05" />

bybash commented on src/ui/src/builder/settings/BuilderFieldsColor.vue:
```diff
@@ -122,13 +122,52 @@ onBeforeUnmount(() => {
 	margin-top: 4px;
 }
+
+.pickerContainer {
```
Where do we use this new class?

rodionchernyshov replied:
it's an UI style improvement
after: 
<img width="362" height="180" alt="Screenshot 2026-01-02 at 15 01 39" src="https://github.com/user-attachments/assets/07d792d7-f725-4a97-8c23-152019839e1d" />
before: 
<img width="313" height="144" alt="Screenshot 2026-01-02 at 15 05 15" src="https://github.com/user-attachments/assets/57dd6709-9e06-4e83-9d05-84535861186f" />

rodionchernyshov committed:
fix: remove unused builderMainWrapper CSS classes

Remove unused .builderMainWrapper and .builderMainWrapper--fullWidth classes
that were defined but never used in the template. The layout uses
mainGrid--fullWidth instead.

Addresses review comments from bybash on PR #1256

rodionchernyshov committed:
revert: restore builderMainWrapper classes for future use

Restore .builderMainWrapper and .builderMainWrapper--fullWidth classes
as they were intended for future use. The classes are currently unused
but kept for planned implementation.

madeindjs commented on src/ui/index.html:
The color scheme changes make the scrollbar black

<img width="1173" height="656" alt="Image" src="https://github.com/user-attachments/assets/0d64b6e9-1d7b-4313-b785-b00ce2c8fd42" />

<img width="1173" height="656" alt="Image" src="https://github.com/user-attachments/assets/ab3ea6fd-9b19-4fe1-a79c-0a07cb7edf8b" />

madeindjs reviewed (changes requested):
The scrollbar switched to black, but it's grey on dev

<img width="1173" height="656" alt="Image" src="https://github.com/user-attachments/assets/0d64b6e9-1d7b-4313-b785-b00ce2c8fd42" />

<img width="1173" height="656" alt="Image" src="https://github.com/user-attachments/assets/ab3ea6fd-9b19-4fe1-a79c-0a07cb7edf8b" />
